numba/core/types/misc.py

```diff
@@ -144,14 +144,15 @@ class CPointer(Type):
     """
     mutable = True
 
-    def __init__(self, dtype):
+    def __init__(self, dtype, addrspace=None):
         self.dtype = dtype
+        self.addrspace = addrspace
         name = "%s*" % dtype
         super(CPointer, self).__init__(name)
```

Comment:
Can you include the addrspace as part of the name? The name is used for symbol mangling. Maybe include the addrspace in name when `addrspace is not None`.

Reply:
I will.

Reply:
Added.